addons/mail/static/tests/suggestion/suggestion.test.js

```diff
@@ -53,6 +53,25 @@ test('display partner mention suggestions on typing "@"', async () => {
     await contains(".o-mail-Composer-suggestion strong", { count: 3 });
 });
 
+test('should display internal users with access when mention suggestions on typing "@"', async () => {
+    const pyEnv = await startServer();
+    const partnerId_1 = pyEnv["res.partner"].create({
+        email: "testpartner@odoo.com",
+        name: "TestPartner",
+        partner_share: true,
+    });
+    const channelId = pyEnv["discuss.channel"].create({
+        name: "dummy channel",
+        channel_member_ids: [Command.create({ partner_id: partnerId_1 })],
+        channel_type: "channel",
+    });
+    await start();
+    await openDiscuss(channelId);
+    await insertText(".o-mail-Composer-input", "@");
+    await contains(".o-mail-Composer-suggestion strong", { count: 2 });
+    await contains(".o-mail-Composer-suggestion strong", { text: "Mitchell Admin" });
+});
```

Comment:
Should create a custom `res.groups` and restrict channel to that group. Have 2 users (new user and `serverState.userId`) in the group, and a 3rd user not in group.

Test that `@` shows Mitchell Admin and the user in group, but not the other. You can also open the invite popover in topbar to see the string "Channel is restricted to 'GroupName'" to give more confidence the channel is configured as expected.

I would also rename test to _"can @user in restricted (group_public_id) channels"_ for clarity.